genie-web/src/main/java/com/netflix/genie/web/aspect/DataServiceRetryAspect.java

```diff
@@ -0,0 +1,115 @@
+/*
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.aspect;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.genie.common.exceptions.GenieException;
+import com.netflix.genie.common.exceptions.GenieServerException;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.Ordered;
+import org.springframework.dao.CannotAcquireLockException;
+import org.springframework.dao.ConcurrencyFailureException;
+import org.springframework.dao.DeadlockLoserDataAccessException;
+import org.springframework.dao.OptimisticLockingFailureException;
+import org.springframework.dao.PessimisticLockingFailureException;
+import org.springframework.dao.QueryTimeoutException;
+import org.springframework.dao.TransientDataAccessResourceException;
+import org.springframework.jdbc.CannotGetJdbcConnectionException;
+import org.springframework.retry.RetryListener;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.stereotype.Component;
+
+import javax.validation.ConstraintViolationException;
+
+/**
+ * Aspect implementation of retrying the data service methods on certain failures.
+ * @author amajumdar
+ * @since 3.0.0
+ */
+@Aspect
+@Component
+@Slf4j
+public class DataServiceRetryAspect implements Ordered {
+    private final RetryTemplate retryTemplate;
+
+    /**
+     * Constructor.
+     * @param noOfRetries number of retries
+     * @param initialInterval initial interval in milliseconds before retrying the first attempt
+     * @param maxInterval maximum interval in milliseconds
+     */
+    @Autowired
+    public DataServiceRetryAspect(
+        @Value("${genie.data.service.retry.noOfRetries:5}") final int noOfRetries,
```

Comment:
Change these to bind to a configuration properties class so we can better respond to code changes.